### PR TITLE
engine: fix potential difference between exchange config name and default

### PR DESCRIPTION
**Pull request by shazbert**
## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

- [ ] ran it

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: engine/sync_manager.go [hunk 1]
```diff
@@ -457,10 +458,11 @@ func (m *syncManager) Update(exchangeName string, p currency.Pair, a asset.Item,
 						createdCounter)
 					m.initSyncWG.Done()
 				}
+				return nil
 			}
 		}
 	}
-	return nil
+	return fmt.Errorf("%w for %s %s %s", errCouldNotSyncNewData, exchangeName, p, a)
```
Comment (gloriousCode) [inline engine/sync_manager.go:465]: I'm not going to ask you to add full coverage of this function, but could you add a test to hit this new error at least?

**Review by gloriousCode — CHANGES_REQUESTED** [review]
I've tested it with a modified config exchange name and it was re-set correctly. I've tested that the error is hit, but one small request from me

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for adding full coverage of the function too!

**Review by thrasher- — APPROVED** [review]
image

tACK!

**Pull request merged**